From: Arnaud Patard <arnaud.patard@rtp-net.org>

[PATCH] Enable write buffer coalescing

RedBoot is disabling write buffer coalescing before giving the hand to
linux. As there's no place in the kernel reenabling it, the performances
are poor (see
http://lists.arm.linux.org.uk/lurker/message/20080308.225920.2ae2af55.en.html).
This patch reenable it.

Signed-off-by: Arnaud Patard <arnaud.patard@rtp-net.org>

---
 arch/arm/mm/proc-xscale.S |    4 	4 +	0 -	0 !
 1 file changed, 4 insertions(+)

Index: linux-2.6/arch/arm/mm/proc-xscale.S
===================================================================
--- linux-2.6.orig/arch/arm/mm/proc-xscale.S	2008-03-09 00:37:05.000000000 +0100
+++ linux-2.6/arch/arm/mm/proc-xscale.S	2008-03-09 12:27:50.000000000 +0100
@@ -114,6 +114,10 @@ clean_addr:	.word	CLEAN_ADDR
  * Nothing too exciting at the moment
  */
 ENTRY(cpu_xscale_proc_init)
+	@ enable write buffer coalescing. Some bootloader disable it
+	mrc	p15, 0, r1, c1, c0, 1
+	bic	r1, r1, #1
+	mcr	p15, 0, r1, c1, c0, 1
 	mov	pc, lr
 
 /*
